src/service/resources-impl.js

```diff
@@ -1723,7 +1723,7 @@ export class Resource {
       return Promise.resolve();
     }
 
-    if (!this.renderOutsideViewport() && !this.isInViewport()) {
+    if (!this.isInViewport() && !this.renderOutsideViewport()) {
```

Comment:
> I think this method is the one that would give us necessary data to calculate this signal.

We could easily do that by returning an enum value instead of a boolean.